### ISSUE TITLE
voltage trace of one compartment Adex cell under step current (0ms -400ms)

### ISSUE THREAD

**Issue by RokasSt** [OpenSourceBrain/VERTEXShowcase#9]
Could you look at this and the trace figure

https://github.com/OpenSourceBrain/VERTEXShowcase/blob/master/VERTEX/Adex_point_neuron_step_current.m

https://github.com/OpenSourceBrain/VERTEXShowcase/blob/master/VERTEX/Adex_point_neuron_step_current_trace1.fig

I wonder whether this is normal that under the current step at 0 ms (lasting for 400ms) there is such a big initial negative phase as in the figure. I cannot reproduce it so far through LEMS, as for the current step at 0ms the initial phase is positive and then 5 APs occur, in contrast to three APs in this VERTEX run. Thanks for help!

@richardtomsett

**Comment by richardtomsett** [comment]
Are you using the non-updated version of VERTEX that initialised the point neuron v_m to zero at the start of the simulation? That would probably explain the negative phase at the start, as it doesn't happen with my version. However I'm surprised the input current actually turned on in your simulation as you've discovered another bug - if you put 0 as the timeOn for the step current, it won't ever switch on because of the way it counts time steps (0 is actually 1 step before the simulation starts...). So I have updated the step current class to take this into account. With the very latest VERTEX in the master branch I am getting a positive initial phase and 4 spikes, with v_m reaching a low of about -110mV due to the AHP current. Not sure why you get 5 spikes with LEMS and 4 spikes with VERTEX though. What integration method does LEMS use? I tried running this with longer and shorter time steps in VERTEX but always got 4 spikes...
@RokasSt

**Comment by RokasSt** [comment]
Hi, Richard
I used the updated version of VERTEX for this with initialization at E_leak ( I fetched your changes). However, I got the same result with the older version, when it still initialized at v_m=0, which prompted me to think what is going on. You use second order Runge-Kutta whereas LEMS use forward Euler. However, I adjusted time step so that errors would be of the same order so it should not be the underlying reason for inconsistency.  I will try to run this simulation with your latest updates in the master branch and I will let you know. Thanks!
@richardtomsett

**Comment by richardtomsett** [comment]
Hmm OK, let me know how it goes. I've tended to find the Izhikevich and AdEx models fire more spikes with forward Euler, especially with longer time steps, but only from my experience- I haven't done a proper analysis or anything. I'm out for the rest of the day but will check back tomorrow/on Monday. Don't work too hard! @RokasSt

**Comment by RokasSt** [comment]
So I ran few simulations with different parameter values and, apparently, there is always the first spike at time zero, which might be responsible for complicating the things; for example, even if the step current is introduced with the time delay 50 ms (timOn=50s), the first spike occurs at time zero. I think it might the reason why it is not consistent with LEMS-based model; this is because after first spike at time zero v is reset and w is updated; however, in the LEMS model with the same parameter values and step current, there is no spike at time zero, v stays on E_leak or other specified value, and the spiking behaviour does not emerge until the step current is introduced. I am not sure whether there was an intention to have that first spike at time zero or this is really a bug. Thanks for help!

@richardtomsett

**Comment by richardtomsett** [comment]
Hmm, I'm not sure why you are getting a spike at time 0. When I run the simulation script with the latest VERTEX update I get four spikes, the first one at 11.59375 ms into the simulation. This is the plot:
![vertex_showcase](https://cloud.githubusercontent.com/assets/1560691/8775763/290da59e-2ee3-11e5-8232-679486068f32.jpg)

@RokasSt

**Comment by RokasSt** [comment]
I ran again the same script I committed and, strangely enough, I get this: 

![adex_test2](https://cloud.githubusercontent.com/assets/12904422/8776400/b391ecf2-2ee8-11e5-875a-c2de49a65053.jpg)

Do you have any ideas why is that? Thanks

@richardtomsett

**Comment by richardtomsett** [comment]
That's an odd one - looks like you are getting the initial spike on the first timestep which causes the hyperpolarisation, and then no more spikes. I think it makes sense that this might happen with the older VERTEX code that started the membrane potential at 0 and didn't apply the input current because the startTime of 0 was before the simulation start. Are you sure you've definitely got the latest VERTEX code from the master github branch rather than the old code in your Matlab path...? @RokasSt

**Comment by RokasSt** [comment]
Hi, Richard
Yes, I even git cloned your master to the new local repository and used it - still the same...  

@richardtomsett

**Comment by richardtomsett** [comment]
That's really strange. I just downloaded the latest master into a new folder on a different computer, ran the script you linked to above from the VERTEX showcase and got exactly the same result as I posted above - four spikes with the first at 11.59375 ms. I'm not really sure what's going on... I don't expect this will shed much light but what version of Matlab/OS are you using? I have tried on Matlab 2015a on Windows and OSX with the same results on each. @RokasSt
